src/server/api/routers/blogPost.ts

```diff
@@ -47,4 +47,31 @@ export const blogPostRouter = createTRPCRouter({
         nextCursor,
       };
     }),
+  getOne: publicProcedure.input(BlogPostFindUniqueSchema).query(({ ctx, input }) => {
+    return ctx.prisma.blogPost.findUnique({
+      where: {
+        id: input.where.id
+      }
+    });
+  }),
+  update: protectedProcedure.input(BlogPostUpdateOneSchema).mutation(async ({ ctx, input }) => {
+      const user = await ctx.prisma.user.findUniqueOrThrow({
+        where: {
+          id: ctx.session.user.id
+        }
+      });
+
+      if (!["CONTRIBUTOR", "ADMIN"].includes(user.role)) {
+        throw new Error("Your account role is not permitted to update blog posts");
+      }
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```

Comment:
Suppressed this for now because I couldn't figure out a clean way to make the front-end data match the generated schema for update in the way that TS wants me to (not sure if this is a result of just TS being overly strict). I would look more into it after MVP, feedback appreciated.

Reply:
it's very weird. It seems like the union of these types is not working as expected
```ts
 export const BlogPostUpdateOneSchema = z.object({
  data: z.union([
    BlogPostUpdateInputObjectSchema,
    BlogPostUncheckedUpdateInputObjectSchema,
  ]),
  where: BlogPostWhereUniqueInputObjectSchema,
});

```
`BlogPostUncheckedUpdateInputObjectSchema` has userId @altarrok any idea about what's causing this?

Reply:
The schema for updates allows the user to provide values at their convenience, so if the user is not updating the user id, then user id field would be undefined, typescript let's you know that and helps you prevent such a bug. Also as I mentioned down below, you shouldn't use the userId provided by the client, to check against session user id. If you still have this problem after those necessary changes, than instead of using the generated schema as input type, I suggest creating your own type that suits the needs of that api, but I don't think it will be necessary.